src/main/java/htsjdk/samtools/util/SortingCollection.java

```diff
@@ -448,19 +463,43 @@ public void remove() {
         private final TreeSet<PeekFileRecordIterator> queue;
 
         MergingIterator() {
-            this.queue = new TreeSet<PeekFileRecordIterator>(new PeekFileRecordIteratorComparator());
+            this.queue = new TreeSet<>(new PeekFileRecordIteratorComparator());
             int n = 0;
-            for (final Path f : SortingCollection.this.files) {
-                final FileRecordIterator it = new FileRecordIterator(f);
+            log.info(String.format("Creating merging iterator from %d files", files.size()));
+            int suggestedBufferSize = checkMemoryAndAdjustBuffer(files.size(), Defaults.BUFFER_SIZE);
+            for (final Path f : files) {
+                final FileRecordIterator it = new FileRecordIterator(f, suggestedBufferSize);
                 if (it.hasNext()) {
                     this.queue.add(new PeekFileRecordIterator(it, n++));
-                }
-                else {
+                } else {
                     it.close();
```

Comment:
This can be skipped if the previous lines throw, using try-with-resources would fix that